src/core/manager.c

```diff
@@ -600,7 +600,7 @@ static int manager_setup_prefix(Manager *m) {
         return 0;
 }
 
-int manager_new(UnitFileScope scope, bool test_run, Manager **_m) {
+int manager_new(UnitFileScope scope, uint8_t test_run, Manager **_m) {
```

Comment:
hmm, let's just make this a generic "unsigned flags" parameter or so. I figure this could sooner or later be useful for the normal invocation too, hence I wouldn't focus this on the "test_run" usecase. Also, uint8_t is a really weird type... I mean, it suggests the size should matter, but it doesn't really, and it's a type that isn't the native integer type on none of the systems we support... Hence, let's be less specific in both cases, let's just call it "flags" and let it be of type "unsigned" and be happy...

Reply:
I made it `uint8_t` to match the field in Manager object. Doesn't matter too much, but a full integer to store two bits seemed a bit wasteful. I can change it to `unsigned` if you prefer.

I called it `test_run` because all the flags that can currently by specified are about testing, so `if (m->test_run)` make sense. If this is changed to just `if (m->flags)`, that's less clear… Dunno, I'll change it if you prefer, but I thought it's slightly clearer with this name.

Reply:
Well, `Manager` is instantiated a single time usually, hence optimizing the size for one 32bit int more or less doesn't really matter I figure.

Also, if you want to optimize that I'd still use "unsigned", but then use `unsigned flags:5` or so when defining `Manager`, which I think is more descriptive what you are doing there, as `:5` is a great way to save some bits, but doesn't leak into the actual type...

Reply:
maybe at least call it `test_run_flags` or `test_run_bits` to clarify that this is a bitset?